### ISSUE TITLE
[feature request] Long & double press support for button mappings

### ISSUE THREAD

**Issue by kennymc-c** [unfoldedcircle/bug-reports#94]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Since the remote just has a limited amount of physical buttons it would be great to map more functions to these buttons via long and double press actions.
This is a common feature in other products like the Logitech Harmony or as a double click on a simple mouse. Even UCRT already makes use of it on the menu button to open the context menu for a page ;)

**Comment by methnen** [comment]
Related to these two:
https://github.com/unfoldedcircle/feature-and-bug-tracker/issues/56
https://github.com/unfoldedcircle/feature-and-bug-tracker/issues/90

Might be they need to be consolidated.

**Comment by kennymc-c** [comment]
No, it's not really related. This feature should allow to map a completely different command from a different device to a long and double press action as it is possible with the Harmony.
But if the integration supports long or double press commands the two commands could be automatically mapped to the same button.

**Comment by methnen** [comment]
Without yours being fulfilled mine and the others are impossible. So I do think they're related.

Someone using an integration inside of an activity for instance would be unable to use any of of the long press actions without your request having been fulfilled because an activities interface has to be customized per activity.

In a general sense the remote currently has no long press functionality, in activities, integrations or otherwise. 

Mostly just wanted to note the connection between them since from an implementation standpoint I think they are connected and from the developer end the work would overlap somewhat.

**Comment by kennymc-c** [comment]
The other issues are about the commands itself. If the integrations would offer these long press actions as a separate command they could also be mapped to a normal short press action. Of course you would then need two separate buttons on the remote but it would still be possible to execute the "long press" command with the integration via a short press action on the remote. So this issue would just makes it easier for the user but it is not absolutely necessary.

**Comment by martonborzak** [comment]
Short press, repeat and long presses are handled by the UI.

We're adding configuration options to the Web Configurator, where one can define a command to be executed for a short press and long press. Double press is not supported.

The current implementation works like this:

If no long press is defined for a button, the short press command will be executed and repeated on press and hold.
If a long press command is defined for a button, the long press command will be executed after a long press timeout. If the button is released before the timeout, the associated short press command will be executed.

**Comment by pr11me** [comment]
Will it be possible to apply long press to the screen buttons?

**Comment by martonborzak** [comment]
> Will it be possible to apply long press to the screen buttons?

Cannot promise, will look into it.

**Comment by pr11me** [comment]
Should I create an issue?

**Comment by WillTriumph1** [comment]
> Short press, repeat and long presses are handled by the UI.
> 
> We're adding configuration options to the Web Configurator, where one can define a command to be executed for a short press and long press. Double press is not supported.
> 
> The current implementation works like this:
> 
> If no long press is defined for a button, the short press command will be executed and repeated on press and hold. If a long press command is defined for a button, the long press command will be executed after a long press timeout. If the button is released before the timeout, the associated short press command will be executed.

Is this a now feature?

**Comment by kennymc-c** [comment]
Tried out 1.7.4 and while long press is working if also a short press command is mapped it doesn't work if you just map a long press command to a button. The remote ui logs the key press but the actual command is not executed.

**Comment by martonborzak** [comment]
> Tried out 1.7.4 and while long press is working if also a short press command is mapped it doesn't work if you just map a long press command to a button. The remote ui logs the key press but the actual command is not executed.

Found the issue, we'll release a patch update today or tomorrow.

**Comment by kennymc-c** [comment]
Thanks, its fixed.
Are there future plans to implement a double press function and long press for soft buttons as this was also mentioned before? Otherwise I would close this ticket if you are not planing to do this.

**Comment by zehnm** [comment]
Long-press is included in 1.7.4.
Double-press mappings are not planned for the near future, unless there's a high demand for it. 

Double-click should normally work with quickly pressing the button twice, but the functionality depends on the device and on timing requirements.

**Comment by kennymc-c** [comment]
If double-click should work on some devices could you explain why it works for the cursor buttons on the Apple-TV but not for Home on the same device? Both should have the same timing requirements.

**Comment by zehnm** [comment]
See #56. It's most likely the long press logic in the UI which prevents a double click.

**Comment by kennymc-c** [comment]
Ok, but then double click does not normally work if the ui long press logic prevents it
